commit 4954d3cda2854670d2540de2f4b6796d58fe7c49
Author: Zihao Yu <yuzihao@ict.ac.cn>
Date:   Sat Nov 9 11:48:13 2019 +0800

    minit: fix assertion fail for qemu
    
    * it seems that qemu can not modify misa

diff --git a/machine/minit.c b/machine/minit.c
index 5ad6d92..317b2b4 100644
--- a/machine/minit.c
+++ b/machine/minit.c
@@ -62,6 +62,15 @@ static void delegate_traps()
   assert(read_csr(medeleg) == exceptions);
 }
 
+static void dump_misa(uint32_t misa) {
+  int i;
+  for (i = 0; i < 26; i ++) {
+    if (misa & 0x1) printm("%c ", 'A' + i);
+    misa >>= 1;
+  }
+  printm("\n");
+}
+
 static void fp_init()
 {
   if (!supports_extension('D') && !supports_extension('F'))
@@ -76,7 +85,8 @@ static void fp_init()
 #else
   uintptr_t fd_mask = (1 << ('F' - 'A')) | (1 << ('D' - 'A'));
   clear_csr(misa, fd_mask);
-  assert(!(read_csr(misa) & fd_mask));
+  dump_misa(read_csr(misa));
+  //assert(!(read_csr(misa) & fd_mask));
 #endif
 }
 
